package.json

```diff
@@ -97,14 +99,206 @@
           "type": "boolean",
           "default": true,
           "description": "Should output channel be raised on formatting error."
+        },
+        "zig.zls.enabled": {
+          "scope": "resource",
+          "type": "boolean",
+          "description": "Whether to enable zls",
+          "default": true
+        },
+        "zig.zls.trace.server": {
+          "scope": "window",
+          "type": "string",
+          "enum": [
+            "off",
+            "messages",
+            "verbose"
+          ],
+          "default": "off",
+          "description": "Traces the communication between VS Code and the language server."
+        },
+        "zig.zls.check_for_update": {
+          "scope": "resource",
+          "type": "boolean",
+          "description": "Whether to automatically check for new updates",
+          "default": true
+        },
+        "zig.zls.path": {
+          "scope": "resource",
+          "type": "string",
+          "description": "Path to `zls` executable. Example: `C:/zls/zig-cache/bin/zig.zls.exe`.",
+          "format": "path"
+        },
+        "zig.zls.debugLog": {
+          "scope": "resource",
+          "type": "boolean",
+          "description": "Enable debug logging in release builds of zig.zls."
+        },
+        "zig.zls.enable_snippets": {
+          "scope": "resource",
+          "type": "boolean",
+          "description": "Enables snippet completions when the client also supports them",
+          "default": false
+        },
+        "zig.zls.enable_ast_check_diagnostics": {
+          "scope": "resource",
+          "type": "boolean",
+          "description": "Whether to enable ast-check diagnostics",
+          "default": true
```

Comment:
If we do go with the ts extension provided ast-check this should be set to false by default or removed entirely.

Reply:
I'll set it to false.

Reply:
If we use zls's ast-check then the extensions ast-check should be disabled when a language server is enabled. This option could also be made more general (`zig.enableAstCheckDiagnostics`) and be used by the extensions ast-check when enabled.

Reply:
[This](https://github.com/ziglang/vscode-zig/pull/78/files#diff-8c9365336029191f4ef1700e71180d7060bf28c4115679a8991044b94c3ef7f3R38) should disable extension diagnostics when zls diagnostics are on. I'll rename the option to something like `zig.astCheckProvider` with enum values of `zls` and `extension` to make it abundantly clear how to switch between both providers.

Reply:
Nice, I forgot about that. The logic just needs to have the and changed to a not .. or (or preferably split into two ifs).

> I'll rename the option to something like `zig.astCheckProvider` with enum values of `zls`, `extension`, and `none` to make it abundantly clear how to switch between both (or neither) providers.

I'm not sure if that level is needed especially if you're confident in zls's ast-check performance but I'll let you decide that.

With those changes I think this will be ready for merging.